pkg/api/validation/validation.go

```diff
@@ -115,6 +117,17 @@ func ValidateAnnotations(annotations map[string]string, fldPath *field.Path) fie
 	if annotations[api.AffinityAnnotationKey] != "" {
 		allErrs = append(allErrs, ValidateAffinityInPodAnnotations(annotations, fldPath)...)
 	}
+
+	hostname := annotations[utilpod.PodHostnameAnnotation]
+	if len(hostname) > 0 && !utilvalidation.IsDNS1123Label(hostname) {
```

Comment:
you should probably do the 2-value return here, and assert that if the label is found, it must be a DNS Label, but if it is not found, ignore.  This will catch any cases of = ""